src/jquery.multiselect.js

```diff
@@ -469,25 +591,47 @@
       var self = this;
       var $button = this.$button;
       function clickHandler() {
-        self[ self._isOpen ? 'close' : 'open' ]();
-        return false;
+         self[ self._isOpen ? 'close' : 'open' ]();
+         return false;
       }
 
       $button
         .on({
           click: clickHandler,
-          keypress: function(e) {
-            switch(e.which) {
-              case 27: // esc
-              case 38: // up
-              case 37: // left
-                self.close();
-                break;
-              case 39: // right
-              case 40: // down
-                self.open();
-                break;
-            }
+          keydown: function(e) {
+             // Change selection via up/down on a closed single select.
+				 if (!self._isOpen && !self.element[0].multiple && e.which in {38:1, 40:1} ) {  // up & down keys
+					var prev, 
+                   current = null, 
+                   next = null;
+					self.$inputs.each( function() {
```

Comment:
This should be a separate method, not a block dropped into the middle of they key handlers.

Reply:
@mlh758 I struggled about what to do with this.   It closes a PR, but it is quite a bit of code to just add specialized up/down key functionality for a closed single select.  I could make a separate method, but it will only be used in the keydown handler.

Reply:
The change itself is welcome, It's just a lot of extra code in them middle of an already large, awkwardly indented block. Putting it into a method like `handleButtonKeyboardNav` and just passing that in would be cleaner here.

Reply:
@mlh758 ok will do